Tensor reinitialization codemod

jerryzh168 opened the pull request:
Summary:
Codemod generated with clangr shard mode, 25 files per diff,
To eliminiate partially initialized Tensor, we split the initialization of local Tensor variables into two steps, first declare un uninitialized Tensor, and
call `ReinitializeTensor` to initialize it.
motivation: https://github.com/pytorch/pytorch/pull/12407

Differential Revision: D13717360

ezyang commented on caffe2/experiments/operators/fully_connected_op_decomposition.h:
```diff
@@ -67,7 +67,7 @@ class FullyConnectedOpDecomp final : public Operator<Context> {
     if (X.dim() > 1) {
       dims = {M, N};
-      multi_buffer_.Resize(M, middle);
+      ReinitializeTensor(&multi_buffer_, {M, middle}, at::dtype<T>().device(Context::GetDeviceType()));
```
`.device(Context::GetDeviceType()` is unnecessary right?

jerryzh168 replied:
`ReinitializeTensor` is a free function, we need to provide device

ezyang commented:
Is there a reason you're not using ghexport to export these diffs? I just realized that this one is stacked (I was wondering why it seemed really long, longer than 25 files)

jerryzh168 commented:
Hmm... this is not meant to be stacked on other diffs in the beginning, so I used the normal export which is more convenient.

jerryzh168 commented on caffe2/experiments/operators/fully_connected_op_prune.h:
```diff
@@ -349,9 +348,9 @@ namespace caffe2 {
       protected:
-       Tensor bias_multiplier_{Context::GetDeviceType()};
+       Tensor bias_multiplier_;
        Tensor sum_buffer_{Context::GetDeviceType()};
```
modified, this is not captured by codemod because it is using ResizeLike

jerryzh168 committed:
Change ConvPoolOp<Context>::SetOutputSize to ConvPoolOp<Context>::GetOutputSize

Summary: Original commit changeset: f1923fdca4a1

Differential Revision: D13885233

fbshipit-source-id: dd1e03848118ebda712f9c14b0b41a8475f80efe

jerryzh168 committed:
Tensor reinitialization codemod (#16378)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/16378

Codemod generated with clangr shard mode, 25 files per diff,
To eliminiate partially initialized Tensor, we split the initialization of local Tensor variables into two steps, first declare un uninitialized Tensor, and
call `ReinitializeTensor` to initialize it.
motivation: https://github.com/pytorch/pytorch/pull/12407

Reviewed By: dzhulgakov

Differential Revision: D13717360

fbshipit-source-id: 53203b0f644a4cbeaf0b3482f6bc1b82a63f0efd
